### ISSUE TITLE
Active/Inactive filter buttons of output dock widget are hard to distinguish in dark mode

### ISSUE THREAD

**Issue by rainman110** [dlr-gtlab/gtlab-core#606]
In GitLab by @schn_bo on Oct 13, 2023, 06:23

## What is the purpose of the change?
- In the black mode of the GTlab the settings of the output are not well distinguishable. It would be great if activated output setting would be stronger emphasis. Moreover, I can not really see a difference between the "Test-Output"-button and the "Show-Hide-Debug-Output"-button. (maybe some spacer between the output visibility and the remaining button could lead to an better useabiltiy.

![grafik](https://my-gitlab-bucket.s3.amazonaws.com/727352741/24db12781c49d1cff2947cfb96e5203581470737ec2103c4b0b10781187653a9/grafik.png)

**Comment by jensschmeink** [comment]
The difference between selected and unselected buttons should be stronger. Icon should be grayed out if it is deactiveted

**Comment by rainman110** [comment]
@mariusalexander Could you take over. I tried to make the color of the icon lighter for the unchecked state. But since you installed a SVG engine, this does not work. I think it will be more efficient, if you look into it.

**Comment by rainman110** [comment]
An alternative would be, to simple use a different SVG for the off-state

**Comment by mariusalexander** [comment]
I think I know a nice solution, I will create a Pull Request soon

**Comment by mariusalexander** [comment]
I have developed the `GtSvgIconEngine` in such a way that one can set custom colors for the normal, disabled and selected mode. In addition there is the `gt::gui::colorize` function that can be used to set custom colors for an icon with these modes:

```cpp
using gt::gui::color::disabled;  // color if button is disabled
using gt::gui::color::highlight; // color if button is enabled
using gt::gui::colorize; // use custom colors for icon

QIcon colorizedIcon = colorize(gt::gui::icon::trace(), highlight, disabled); // takes the function pointers for enabled/disabled 
```

However, checked `QPushButtons` do not use the active/disabled icons for when the button is (not) checked. QIcon can also distinguish between its state (On, Off) [see QIcon documentation](https://doc.qt.io/qt-6/qicon.html#making-classes-that-use-qicon). However, the icon engine does not respect the state yet. It is an easy fix though, but I am not so sure whether this would cause ABI changes. I tried changing it locally, but it seems that checked `QPushButtons` also does **not** use icons with their state set to On/Off ... :( So even if the icon engine would support that, it would not help here. At the end I still had to implement an alternative, not so neat solution after all.
